packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -129,12 +129,41 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     return super.validateCommand();
   }
 
+  void _validateXcodeBuildSettings() {
+    final Map<String, String>? buildSettings = xcodeBuildResult?.xcodeBuildExecution?.buildSettings;
+    if (buildSettings == null) {
+      return;
+    }
+
+    final Map<String, String?> xcodeProjectSettingsMap = <String, String?>{};
+
+    xcodeProjectSettingsMap['iOS App Version'] = buildSettings['MARKETING_VERSION'];
+    xcodeProjectSettingsMap['iOS Build Number'] = buildSettings['CURRENT_PROJECT_VERSION'];
+    xcodeProjectSettingsMap['App Display Name'] = buildSettings['INFOPLIST_KEY_CFBundleDisplayName'];
```

Comment:
You can rely on any of these build settings being present, and actually in a newly created app `MARKETING_VERSION` and `INFOPLIST_KEY_CFBundleDisplayName` aren't.  Much more reliable to parse the built app's Info.plist, which is what the OS uses.
```
$ plutil -p build/ios/iphoneos/Runner.app/Info.plist
  "CFBundleDisplayName" => "Test Create"
  "CFBundleIdentifier" => "com.example.testCreate"
  "CFBundleShortVersionString" => "100.0.0"
  "CFBundleVersion" => "200"
  "MinimumOSVersion" => "11.0"
```

https://github.com/flutter/flutter/issues/97730#issuecomment-1270766299

Reply:
Yes, it's created by Xcode based on those inputs, as well as things like what version of Xcode was used to build it.  All app bundles have a generated Info.plist like this.